sw/device/tests/meson.build

```diff
@@ -5,25 +5,79 @@
 sw_tests = {}
 
 subdir('dif')
-subdir('flash_ctrl')
-subdir('hmac')
-subdir('rv_timer')
-subdir('consecutive_irqs')
 
 aes_test_lib = declare_dependency(
   link_with: static_library(
     'aes_test_lib',
     sources: ['aes_test.c'],
     dependencies: [
       sw_lib_aes,
-      sw_lib_uart,
+      sw_lib_base_log,
+    ],
+  ),
+)
+
+consecutive_irqs_test_lib = declare_dependency(
+  link_with: static_library(
+    'consecutive_irqs_test_lib',
+    sources: ['consecutive_irqs_test.c'],
+    dependencies: [
+      dif_uart,
+      dif_plic,
+      sw_lib_irq,
+      sw_lib_mmio,
+      sw_lib_base_log,
+      sw_lib_runtime_hart,
+      sw_lib_testing_test_status,
+    ],
+  ),
+)
+
+flash_ctrl_test_lib = declare_dependency(
+  link_with: static_library(
+    'flash_ctrl_test_lib',
+    sources: ['flash_ctrl_test.c'],
+    dependencies: [
       sw_lib_mem,
+      sw_lib_mmio,
+      sw_lib_flash_ctrl,
+      sw_lib_base_log,
+    ],
+  ),
+)
+
+rv_timer_test_lib = declare_dependency(
+  link_with: static_library(
+    'rv_timer_test_lib',
+    sources: ['rv_timer_test.c'],
+    dependencies: [
+      sw_lib_irq,
+      sw_lib_pinmux,
+      sw_lib_rv_timer,
+      sw_lib_base_log,
+      sw_lib_dif_gpio,
+    ],
+  ),
+)
+
+sha256_test_lib = declare_dependency(
+  link_with: static_library(
+    'sha256_test_lib',
+    sources: ['sha256_test.c'],
+    dependencies: [
+      sw_lib_hmac,
+      sw_lib_flash_ctrl,
+      sw_lib_base_log,
     ],
   ),
 )
 
 sw_tests += {
```

Comment:
Is there a reason this needs to be a `+=`?

Reply:
In case we decide to add subdirs in future.

Reply:
Fair.